File: NaSt2D-2.0/Tools/f/delete.f
C****************************************************************************
C     Program to delete of repeately occuring data points                   
C     -1. -1. -1. labelled the input end                                    
C---------------------------------------------------------------------------
C     Programm zum Loeschen mehrfach hintereinander liegender Datenpunkte   
C     -1. -1. -1. kennzeichnet das Eingabeende                              
C****************************************************************************
      program Delete
c
      parameter (n=10000)
      dimension a(n),b(n),c(n)
      open (unit=1,file='ein')
      open (unit=2,file='aus')
c
      do 10 i=1,n
      read(1,*) a(i),b(i),c(i) 
      if(a(i).eq.-1..and.b(i).eq.-1..and.c(i).eq.-1.) goto 20
   10 continue
   20 in=i-1
      write(2,*) a(1),b(1),c(1)

      in=in+1
      do 30 i=2,in
   30 if(a(i).ne.a(i-1).or.b(i).ne.b(i-1)) write(2,*) a(i),b(i),c(i)
 
      stop 'ok'
      end
c*******************************************************************

File: NaSt2D-2.0/Tools/c/delete.c
/*     delete.c */
/*****************************************************************************/
/*     Program to delete of repeately occuring data points                   */
/*     -1. -1. -1. labelled the input end                                    */
/*---------------------------------------------------------------------------*/
/*     Programm zum Loeschen mehrfach hintereinander liegender Datenpunkte   */
/*     -1. -1. -1. kennzeichnet das Eingabeende                              */
/*****************************************************************************/
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

#define VOL1 1000   /* 1st dimension row    - Zeilenindex  i */
#define VOL2 3      /* 2nd dimension column - Spaltenindex j */

int main(void) 
{
   int    i=0,in=0,j=0;
   float  array[VOL1][VOL2];
   FILE   *fpein;
   FILE   *fpaus;
   char   *input="ein";                            /* input file  */
   char   *output="aus";                           /* output file */

   fpein=fopen(input,"r");                         /* opening the files */
   fpaus=fopen(output,"w");
   if ((fpein==NULL)||(fpaus==NULL)) {
      printf("Error on opening the files\n");
      return(-1);                                  /* EXIT_ERROR_input_file */
                                     }

   for(i=0; i < VOL1; i++) {                       /* read in array */
      for(j=0; j < VOL2; j++) fscanf(fpein,"%f ", &array[i][j]);
      in=i-1;
      if(array[i][0] == array[i][1] && array[i][0] == array[i][2]) break;
                           }
   in=in+1;

   for(j=0; j < VOL2; j++) fprintf(fpaus,"%3.8f ", array[0][j]);	
   fprintf(fpaus,"\n");
   for(i=1; i <= in; i++) {                       /* delete repeated occurrence */	
	if(array[i][0] != array[i-1][0] || array[i][1] != array[i-1][1]) { 						
	    for(j=0; j < VOL2; j++) fprintf(fpaus,"%3.8f ", array[i][j]);								
	    fprintf(fpaus,"\n");																	
                                                                     	 }	  
	                  }
							 
   printf("delete EXIT_SUCCESS\n");
   fclose(fpein);
   fclose(fpaus);
   return EXIT_SUCCESS;
}














